txcache/txCache.go

```diff
@@ -99,7 +99,12 @@ func (cache *TxCache) GetByTxHash(txHash []byte) (*WrappedTransaction, bool) {
 
 // SelectTransactions selects the best transactions to be included in the next miniblock.
 // It returns up to "maxNum" transactions, with total gas <= "gasRequested".
-func (cache *TxCache) SelectTransactions(gasRequested uint64, maxNum int) ([]*WrappedTransaction, uint64) {
+func (cache *TxCache) SelectTransactions(accountStateProvider AccountStateProvider, gasRequested uint64, maxNum int, selectionLoopMaximumDuration time.Duration) ([]*WrappedTransaction, uint64) {
```

Comment:
Now, we also pass a wrapped accounts adapter, so that we can learn the nonce, balance etc. of a sender with pending transactions. This allows us to perform a much more informed selection.

Additionally, we parametrize the selection with `selectionLoopMaximumDuration` (safety mechanism).